Week-13/homework-13/homework13.js

```diff
@@ -0,0 +1,162 @@
+      { text: '1994', correct: false },
+      { text: 'none of the above', correct: false },
+    ],
+  },
+  {
+    question: 'Which tool can you use to ensure code quality?',
+    answers: [
+      { text: 'Angular', correct: false },
+      { text: 'jQuery', correct: false },
+      { text: 'RequireJS', correct: false },
+      { text: 'ESLint', correct: true },
+    ],
+  },
+];
+
+//For video referening --> myNames = videoNames
+//quizData = questions
+//questionElement = questionElement
+//answerButtons = answerButtons
+//nextButton = nextButton
+
+// Selecting the question, question number, answer buttons, and next button
+let questionElement = document.querySelector(`[data-component="question"]`);
+let questionNumber = document.querySelector(
+  `[data-component="questionNumber"]`
+);
+let answerButtons = document.querySelector(`[data-component="answers"]`);
+let nextButton = document.querySelector(`[data-component="next"]`);
+
+// Setting some index for this medthod?
+let currentQuestionIndex = 0;
+let score = 0;
+
+//Im guessing a function when starting the page, also adds "NeXT" into nextButton
+function startQuiz() {
+  currentQuestionIndex = 0;
+  nextButton.innerHTML = 'Next';
+  score = 0;
+  showQuestion();
+}
+
+//Creating some function that would load corresponding question number and question
+function showQuestion() {
+  resetState();
+
+  let currentQuestion = quizData[currentQuestionIndex];
+  let questionNumber = currentQuestionIndex + 1;
+
+  questionElement.innerHTML =
+    'Question ' + questionNumber + '. ' + currentQuestion.question;
+
+  //Adding the answers as buttons
+  currentQuestion.answers.forEach((answer) => {
+    let button = document.createElement('button');
+    button.innerHTML = answer.text;
+    button.classList.add('btn');
+    button.classList.add('btn-outline-secondary');
+    button.classList.add('d-block');
+    answerButtons.appendChild(button);
+    //Adding click event when click an answer (also don't understand what "dataset" is)
+    if (answer.correct) {
+      button.dataset.correct = answer.correct;
+    }
+    button.addEventListener('click', selectAnswer);
+  });
+}
+
+//Adding a reset function - when moving to new question it moves previous asnwers
+function resetState() {
+  nextButton.style.display = 'none';
+  while (answerButtons.firstChild) {
+    answerButtons.removeChild(answerButtons.firstChild);
+  }
+}
+
+//Adding function when selecting an answer - display correct/incorrect answers
+function selectAnswer(e) {
+  //is there a difference between e.target and event.target?
+  let selectBtn = e.target;
+  let isCorrect = selectBtn.dataset.correct;
+  if (isCorrect === 'true') {
+    selectBtn.classList.add('btn-success');
+    //Increase the score by one when correct
+    score++;
+  } else {
+    selectBtn.classList.add('btn-danger');
+  }
+  //Not sure what it's adding here - need to revisit!!!
+  Array.from(answerButtons.children).forEach((button) => {
```

Comment:
Interesting, another usage of ChatGPT, it's good that you try to find the answer from ChatGPT, however, I would strongly recommend using our generic `for...loop` for Array first